src/main/webapp/app/tasks/default-task.service.ts

```diff
@@ -23,4 +23,12 @@ export class DefaultTaskService implements TaskService {
   getAll(): Observable<Task[]> {
     return this.http.get<Task[]>(this.baseUrl + '/tasks');
   }
+
+  updateStatus(id: string, status: boolean): Observable<void> {
+    return this.http.patch<void>(`${this.baseUrl}/tasks/${id}`, { completed: status });
```

Comment:
Why the use of patch over put (for example)?

Reply:
When ever we are replacing the resource entirely, we can use PUT. When we doing a partial update, we can use HTTP PATCH. Here in this scenario we are jus updating completed status of a task. So PATCH is preferable.